ReactAndroid/src/main/java/com/facebook/react/ReactFragment.java

```diff
@@ -0,0 +1,289 @@
+      getReactNativeHost().getReactInstanceManager().onHostResume(getActivity(), this);
+    }
+  }
+
+  @Override
+  public void onPause() {
+    super.onPause();
+    if (getReactNativeHost().hasInstance()) {
+      getReactNativeHost().getReactInstanceManager().onHostPause(getActivity());
+    }
+  }
+
+  @Override
+  public void onDestroy() {
+    super.onDestroy();
+    if (mReactRootView != null) {
+      mReactRootView.unmountReactApplication();
+      mReactRootView = null;
+    }
+    if (getReactNativeHost().hasInstance()) {
+      getReactNativeHost().getReactInstanceManager().onHostDestroy(getActivity());
+    }
+  }
+  // endregion
+
+  /**
+   * This currently only checks to see if we've enabled the permission to draw over other apps.
+   * This is only used in debug/developer mode and is otherwise not used.
+   *
+   * @param requestCode Code that requested the activity
+   * @param resultCode  Code which describes the result
+   * @param data        Any data passed from the activity
+   */
+  @Override
+  public void onActivityResult(int requestCode, int resultCode, Intent data) {
+    super.onActivityResult(requestCode, resultCode, data);
+    if (requestCode == REQUEST_OVERLAY_CODE) {
+      if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+        if (Settings.canDrawOverlays(getContext())) {
+          mReactRootView.startReactApplication(
+            getReactNativeHost().getReactInstanceManager(),
+            mComponentName,
+            mLaunchOptions);
+          Toast.makeText(getContext(), REDBOX_PERMISSION_GRANTED_MESSAGE, Toast.LENGTH_LONG).show();
+        }
+      }
+    }
+  }
+
+  /**
+   * Helper to forward hardware back presses to our React Native Host
+   *
+   * This must be called via a forward from your host Activity
+   *
+   */
+  public void onBackPressed() {
+    if (getReactNativeHost().hasInstance()) {
+      getReactNativeHost().getReactInstanceManager().onBackPressed();
+    }
+  }
+
+  /**
+   * Helper to forward onKeyUp commands from our host Activity.
+   * This allows ReactFragment to handle double tap reloads and dev menus
+   *
+   * This must be called via a forward from your host Activity
+   *
+   * @param keyCode keyCode
+   * @param event   event
+   * @return true if we handled onKeyUp
+   */
+  public boolean onKeyUp(int keyCode, KeyEvent event) {
+    boolean handled = false;
+    if (getReactNativeHost().getUseDeveloperSupport() && getReactNativeHost().hasInstance()) {
+      if (keyCode == KeyEvent.KEYCODE_MENU) {
+        getReactNativeHost().getReactInstanceManager().showDevOptionsDialog();
+        handled = true;
+      }
+      boolean didDoubleTapR = Assertions.assertNotNull(mDoubleTapReloadRecognizer).didDoubleTapR(keyCode, getActivity().getCurrentFocus());
+      if (didDoubleTapR) {
+        getReactNativeHost().getReactInstanceManager().getDevSupportManager().handleReloadJS();
+        handled = true;
+      }
+    }
+    return handled;
```

Comment:
This looks like copy & paste? Can you extract it to a shared helper rather than copy & pasting?